include/range/v3/utility/common_iterator.hpp

```diff
@@ -27,6 +31,105 @@ namespace ranges
         /// \cond
         namespace detail
         {
+#if defined(__GNUC__) && !defined(__clang__) && __GNUC__ == 6 && __GNUC_MINOR__ == 2
+            // Semi-workaround https://gcc.gnu.org/bugzilla/show_bug.cgi?id=77537
+            template<typename T, typename... Args>
+            struct pair_construct_hack
+              : Constructible<T, Args...>
+            {};
+
+            template<typename First, typename Second, typename Arg1, typename Arg2>
+            struct pair_construct_hack<std::pair<First, Second>, std::pair<Arg1, Arg2>>
+              : meta::and_<
+                    std::is_constructible<First, Arg1>,
+                    std::is_constructible<Second, Arg2>,
+                    meta::defer<Constructible, std::pair<First, Second>, std::pair<Arg1, Arg2>>>
+            {};
+
+            template<typename First, typename Second, typename Arg1, typename Arg2>
+            struct pair_construct_hack<std::pair<First, Second>, common_pair<Arg1, Arg2>>
+              : meta::and_<
+                    std::is_constructible<First, Arg1>,
+                    std::is_constructible<Second, Arg2>,
+                    meta::defer<Constructible, std::pair<First, Second>, std::pair<Arg1, Arg2>>>
+            {};
+
+            template<typename T, typename... Args>
+            using is_nothrow_constructible_hack = meta::and_<
+                pair_construct_hack<T, Args...>,
+                meta::defer<std::is_nothrow_constructible, T, Args...>>;
+#else
+            template<typename T, typename... Args>
+            using pair_construct_hack = Constructible<T, Args &&...>;
+
+            template<typename T, typename... Args>
+            using is_nothrow_constructible_hack = std::is_nothrow_constructible<T, Args...>;
+#endif
+
+            template<typename T>
+            class operator_arrow_proxy
+            {
+                T value_;
+            public:
+                template<typename U, CONCEPT_REQUIRES_(pair_construct_hack<T, U>())>
```

Comment:
The hack is here to make `view::zip(...) | view::bounded` work, specifically on test/view/zip.cpp line 214. Now that 6.3 is released - with PR 77537 fixed - I'm hoping the problem will die soon and we'll be able to ignore the fact that unpatched 6.2 is broken.